dotnet/src/Extensions/Reliability.Polly/Config/DefaultHttpRetryHandler.cs

```diff
@@ -0,0 +1,135 @@
+    /// <returns>Returns the async policy executor</returns>
+    private AsyncPolicy<HttpResponseMessage> GetPolicy()
+    {
+        if (this.IsRetryDisabled())
+        {
+            return Policy.NoOpAsync<HttpResponseMessage>();
+        }
+
+        var policyBuilder = Policy.HandleResult<HttpResponseMessage>(response =>
+            {
+                return this._config.RetryableStatusCodes.Contains(response.StatusCode);
+            });
+
+        policyBuilder = policyBuilder.Or<Exception>(e => this._config.RetryableExceptionTypes.Contains(e.GetType()));
+
+        var timeoutPolicy = Policy.TimeoutAsync<HttpResponseMessage>(this._config.MaxTotalRetryTime);
+
+        var retryPolicy = policyBuilder.WaitAndRetryAsync(
+            retryCount: this._config.MaxRetryCount,
+            sleepDurationProvider: (retryCount, context, timeSpan) =>
+            {
+                return this.GetWaitTime(retryCount, context.Result);
+            },
+            (outcome, timespan, retryCount, _) =>
+            {
+                this._logger?.LogWarning(
+                    "Error executing action [attempt {CurrentRetryAttempt} of {MaxRetryAttempts}]. Reason: {StatusCode}. Will retry after {WaitMilliseconds}ms.",
+                    retryCount,
+                    this._config.MaxRetryCount,
+                    outcome.Result.StatusCode,
+                    timespan.TotalMilliseconds);
+                return Task.CompletedTask;
+            });
+
+        return Policy.WrapAsync<HttpResponseMessage>(timeoutPolicy, retryPolicy);
+    }
+
+    private bool IsRetryDisabled()
+    {
+        return this._config.MaxRetryCount == 0;
+    }
+
+    /// <summary>
+    /// Get the wait time for the next retry.
+    /// </summary>
+    /// <param name="retryCount">Current retry count</param>
+    /// <param name="response">The response message that potentially contains RetryAfter header.</param>
+    private TimeSpan GetWaitTime(int retryCount, HttpResponseMessage? response)
+    {
+        // If the response contains a RetryAfter header, use that value
+        // Otherwise, use the configured min retry delay
+        var retryAfter = response?.Headers.RetryAfter?.Date.HasValue == true
+            ? response?.Headers.RetryAfter?.Date - this._timeProvider.GetCurrentTime()
+            : (response?.Headers.RetryAfter?.Delta) ?? this._config.MinRetryDelay;
+        retryAfter ??= this._config.MinRetryDelay;
+
+        // If the retry delay is longer than the max retry delay, use the max retry delay
+        var timeToWait = retryAfter > this._config.MaxRetryDelay
+            ? this._config.MaxRetryDelay
+            : retryAfter < this._config.MinRetryDelay
+                ? this._config.MinRetryDelay
+                : retryAfter ?? default;
+
+        // If exponential backoff is enabled, double the delay for each retry
+        if (this._config.UseExponentialBackoff)
+        {
+            timeToWait = TimeSpan.FromTicks(timeToWait.Ticks * 2);
```

Comment:
1. This line only doubles the wait time instead of the intended exponential increase, which can be achieved using the `retryCount` parameter.

2. I'm not sure if we should exponentially increase the wait time for responses with `RetryAfter.Date` or `RetryAfter.Delta` headers. It makes sense for responses without a specified wait time, but if the server already provides a retry instruction, why increase it exponentially?

Reply:
1. Agree, this was the current default implementation in the Core Kernel. Changed.

2. You are right, good catch, I left this part with the original implementation.

Reply:
> This line only doubles the wait time instead of the intended exponential increase

How would it be different? My understanding is that exponential increase is when you take the original delay and multiply by a factor of x>=2. When initial wait is 1 second, you get subsequent waits of 2, 4, 8, 16, 32. This is exponential growth.

Reply:
> > This line only doubles the wait time instead of the intended exponential increase
> 
> How would it be different? My understanding is that exponential increase is when you take the original delay and multiply by a factor of x>=2. When initial wait is 1 second, you get subsequent waits of 2, 4, 8, 16, 32. This is exponential growth.

Yep, your understanding is correct; the subsequent waits should be 2, 4, 8, 16, 32 for exponential retry. However, with the current implementation where the `retryCount` parameter is not used, the waits will remain at 2, 2, 2, 2, 2, ... I think the line should use the `retryCount` to achieve that - `timeToWait = TimeSpan.FromTicks(timeToWait.Ticks * Math.Pow(2, retryCount));`